### PR TITLE
Do not use silencer scalac params when running doc task

### PR DESCRIPTION
**Pull request by seglo**
## Purpose

The travis build fails when publishing artifacts for google-cloud-pub-sub because the `doc` task doesn't have the silencer plugin in scope.  For some reason this is only an issue in the 2.12 and 2.13 cross builds.

Ex)

```
[error] (google-cloud-pub-sub-grpc / Compile / doc) Scaladoc generation failed
```

https://travis-ci.com/akka/alpakka/jobs/273152362#L1611
https://travis-ci.com/akka/alpakka/jobs/273152363#L2199

## References

* Issue created in #2050
* Ref: https://github.com/ghik/silencer/issues/40

## Changes

* Only apply silencer `scalac` parameters in `Compile, compile` targets.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -169,7 +169,7 @@ lazy val googleCloudPubSubGrpc = alpakkaProject(
   javaAgents += Dependencies.GooglePubSubGrpcAlpnAgent % Test,
   // for the ExampleApp in the tests
   connectInput in run := true,
-  scalacOptions += "-P:silencer:pathFilters=src_managed",
+  scalacOptions in (Compile, compile) += "-P:silencer:pathFilters=src_managed",
```
Comment (ennru) [inline build.sbt]: I suggest to use the `Compile / compile / scalacOptions` notation, as we do in most places now.

**Comment by seglo** [comment]
Qualifying the scope to `Compile` will skip applying the parameter for `Test` scope.  Instead, I've added a filter to remove the parameter when running the `doc` task, as mentioned in https://github.com/ghik/silencer/issues/40 .

**Comment by seglo** [comment]
Transient failures
* https://github.com/akka/alpakka/issues/2073
* https://github.com/akka/alpakka/issues/1549

**Comment by seglo** [comment]
After I merged I found that this can be simplified by omitting the `intask` (`compile`) part of the scoping.

```
Compile / scalacOptions += "-P:silencer:pathFilters=src_managed"
```

**Pull request merged**